### PR TITLE
PANW AutoFocus V2 indicators context outputs - Support for version 5.0

### PR DESCRIPTION
**Pull request by guyfreund**
## Status
Ready

## Related Issues
fixes: https://github.com/demisto/etc/issues/17949

## Description
Updated Palo Alto Networks AutoFocus V2 Indicators context outputs to support version 5.0.

## Does it break backward compatibility?
   - No

## Must have
- [ ] Tests
- [ ] Documentation (with link to it)
- [ ] Code Review

### CODE REVIEW

**Comment by guyfreund** [comment]
Hey @kirbles19, the following files were edited:
- [AutoFocusV2.yml](https://github.com/demisto/content/blob/5862f7f2c3ffca74d802bd74101d61497524d156/Integrations/AutofocusV2/AutofocusV2.yml)
- [CHANGELOF.md](https://github.com/demisto/content/blob/5862f7f2c3ffca74d802bd74101d61497524d156/Integrations/AutofocusV2/CHANGELOG.md)

Commands that have been changed:
1. Command "autofocus-samples-search-results" - Context outputs added to the yml file:
1.1  File.Size
1.2 File.SHA1
1.3 File.SHA256
1.4 File.Type
1.5 File.Tags
2. Command "autofocus-sessions-search-results" - Context outputs added to the yml file:
2.1 File.Name
2.2 File.SHA256
2.3 File.Tags
3. Command "get-session-details" - Context outputs added to the yml file:
3.1 File.Name
3.2 File.SHA256

File: Integrations/AutofocusV2/AutofocusV2.py [hunk 1]
```diff
@@ -522,6 +529,25 @@ def print_hr_by_category(category_name, category_data):
             })
 
 
+def get_file_context_indicators(results):
+    files = []
+    for result in results:
+        raw_file = get_fields_from_hit_object(result, 'file_indicators')
+        file = get_relevant_fields(raw_file, 'file_indicators')
```
Comment (Itay4) [inline Integrations/AutofocusV2/AutofocusV2.py]: dont use reserved words as var names, use `file_` for example

File: Integrations/AutofocusV2/AutofocusV2.py [hunk 2]
```diff
@@ -522,6 +529,25 @@ def print_hr_by_category(category_name, category_data):
...
+        files.append(file)
+    return files
+
+
+def get_relevant_fields(result_object, response_dict_name):
```
Comment (Itay4) [inline Integrations/AutofocusV2/AutofocusV2.py]: improve func name

File: Integrations/AutofocusV2/AutofocusV2.py [hunk 3]
```diff
@@ -522,6 +529,25 @@ def print_hr_by_category(category_name, category_data):
...
+    af_params_dict = API_PARAM_DICT.get(response_dict_name)
+    new_object = {}
```
Comment (Itay4) [inline Integrations/AutofocusV2/AutofocusV2.py]: improve dict name

File: Integrations/AutofocusV2/AutofocusV2.py (hunk 2 above)
Comment (Itay4) [inline Integrations/AutofocusV2/AutofocusV2.py]: add docstring and add example of input and output

File: Integrations/AutofocusV2/AutofocusV2.py [hunk 4]
```diff
@@ -593,7 +620,8 @@ def samples_search_results_command():
         'Contents': results,
         'EntryContext': {'AutoFocus.SamplesResults(val.ID == obj.ID)': results,
```
Comment (Itay4) [inline Integrations/AutofocusV2/AutofocusV2.py]: need to import linting here
build the entry context object before calling `demisto.results` and pass it here

File: Integrations/AutofocusV2/AutofocusV2.py [hunk 5]
```diff
@@ -612,7 +641,8 @@ def sessions_search_results_command():
         'Contents': results,
         'EntryContext': {'AutoFocus.SessionsResults(val.ID == obj.ID)': results,
                          'AutoFocus.SessionsSearch(val.AFCookie == obj.AFCookie)': {'Status': status,
-                                                                                    'AFCookie': af_cookie}},
```
Comment (Itay4) [inline Integrations/AutofocusV2/AutofocusV2.py]: same here

File: Integrations/AutofocusV2/AutofocusV2.py [hunk 6]
```diff
@@ -621,12 +651,14 @@ def get_session_details_command():
...
     session_id = args.get('session_id')
     result = get_session_details(session_id)
+    file = get_file_context_indicators(result)
     md = tableToMarkdown(f'Session {session_id}:', result)
     demisto.results({
         'Type': entryTypes['note'],
         'ContentsFormat': formats['text'],
         'Contents': result,
-        'EntryContext': {'AutoFocus.Sessions(val.ID == obj.ID)': result},
+        'EntryContext': {'AutoFocus.Sessions(val.ID == obj.ID)': result,
```
Comment (Itay4) [inline Integrations/AutofocusV2/AutofocusV2.py]: same here

**Comment by richardbluestone** [comment]
@guyfreund I couldn't find these Context outputs in yml file for the autofocus-samples-search-results" command:
1.1 File.Size
1.2 File.SHA1
1.3 File.SHA256
1.4 File.Type (I found this one)
1.5 File.Tags

I found them without the "file" output, so for example, AutoFocus.SamplesResults.SHA256.

**Comment by guyfreund** [comment]
> @guyfreund I couldn't find these Context outputs in yml file for the autofocus-samples-search-results command:
> 1.1 File.Size
> 1.2 File.SHA1
> 1.3 File.SHA256
> 1.4 File.Type (I found this one)
> 1.5 File.Tags
> 
> I found them without the "file" output, so for example, AutoFocus.SamplesResults.SHA256.
> Can you let me know what you want me to review?

I've updated it now.

**Comment by richardbluestone** [comment]
Hi, I've reviewed the md and yml files.

File: Integrations/AutofocusV2/AutofocusV2.py [hunk 7]
```diff
@@ -522,6 +529,63 @@ def print_hr_by_category(category_name, category_data):
...
+                            }
+    output: {
+                "SHA256": "18c9acd34a3aea09121f027857e0004a3ea33a372b213a8361e8a978330f0dc8",
+                "Name": "wildfire-test-pe-file.exe"
+            }
+    :param result_object: a dictionary representing an object
+    :param response_dict_name: a dictionary which it's values are the relevant fields (filters)
+    :return: the result_object filtered by the relevant fields
+    """
+    print(result_object)
```
Comment (Itay4) [inline Integrations/AutofocusV2/AutofocusV2.py]: remove this print

File: Integrations/AutofocusV2/AutofocusV2.py [hunk 8]
```diff
@@ -583,17 +647,21 @@ def samples_search_results_command():
     args = demisto.args()
     af_cookie = args.get('af_cookie')
     results, status = get_search_results('samples', af_cookie)
+    files = get_files_data_from_results(results)
     if len(results) < 1:
         md = results = 'No entries found that match the query'
     else:
         md = tableToMarkdown(f'Search Samples Results is {status}', results)
+    context = {
+        'AutoFocus.SamplesResults(val.ID == obj.ID)': results,
```
Comment (Itay4) [inline Integrations/AutofocusV2/AutofocusV2.py]:
```suggestion
        'AutoFocus.SamplesResults(val.ID === obj.ID)': results,
```

File: Integrations/AutofocusV2/AutofocusV2.py [hunk 9]
```diff
@@ -583,17 +647,21 @@ def samples_search_results_command():
...
+        'AutoFocus.SamplesSearch(val.AFCookie == obj.AFCookie)': {'Status': status, 'AFCookie': af_cookie},
```
Comment (Itay4) [inline Integrations/AutofocusV2/AutofocusV2.py]:
```suggestion
        'AutoFocus.SamplesSearch(val.AFCookie === obj.AFCookie)': {'Status': status, 'AFCookie': af_cookie},
```

File: Integrations/AutofocusV2/AutofocusV2.py [hunk 10]
```diff
@@ -602,17 +670,21 @@ def sessions_search_results_command():
     args = demisto.args()
     af_cookie = args.get('af_cookie')
     results, status = get_search_results('sessions', af_cookie)
+    files = get_files_data_from_results(results)
     if len(results) < 1:
         md = results = 'No entries found that match the query'
     else:
         md = tableToMarkdown(f'Search Samples Results is {status}', results)
+    context = {
+        'AutoFocus.SessionsResults(val.ID == obj.ID)': results,
```
Comment (Itay4) [inline Integrations/AutofocusV2/AutofocusV2.py]:
```suggestion
        'AutoFocus.SessionsResults(val.ID === obj.ID)': results,
```

File: Integrations/AutofocusV2/AutofocusV2.py [hunk 11]
```diff
@@ -602,17 +670,21 @@ def sessions_search_results_command():
...
+        'AutoFocus.SessionsSearch(val.AFCookie == obj.AFCookie)': {'Status': status, 'AFCookie': af_cookie},
```
Comment (Itay4) [inline Integrations/AutofocusV2/AutofocusV2.py]:
```suggestion
        'AutoFocus.SessionsSearch(val.AFCookie === obj.AFCookie)': {'Status': status, 'AFCookie': af_cookie},
```

File: Integrations/AutofocusV2/AutofocusV2.py [hunk 12]
```diff
@@ -621,12 +693,17 @@ def get_session_details_command():
     args = demisto.args()
     session_id = args.get('session_id')
     result = get_session_details(session_id)
+    files = get_files_data_from_results(result)
     md = tableToMarkdown(f'Session {session_id}:', result)
+    context = {
+        'AutoFocus.Sessions(val.ID == obj.ID)': result,
```
Comment (Itay4) [inline Integrations/AutofocusV2/AutofocusV2.py]:
```suggestion
        'AutoFocus.Sessions(val.ID === obj.ID)': result,
```

**Pull request merged**